packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerPlugin.java

```diff
@@ -21,6 +21,9 @@
   private final ImagePickerDelegate delegate;
 
   public static void registerWith(PluginRegistry.Registrar registrar) {
+    if (registrar.activity() == null) {
```

Comment:
This will prevent the crash but IIUC leaves us with an unregistered plugin, as even if the activity asks the GeneratedPluginRegistrant to register again it will short circuit.

Unless I'm missing something this will trade the startup crash with a crash the first time the plugin is used (as the method channel isn't registered).

I believe the only eager use of activity is getExternalFilesDir and this can be done with the application context which is guaranteed to be available.
It looks like ImagePickerDelegate is caching the activity that's being passed to it from here, I believe instead of doing that we can make ImagePickerDelegate always ask the activity from the registrar when it needs an activity.

@cyanglaz @Hixie am I getting this wrong? is the current fix actually leaving the picker in a working state? are you interested in merging this for now just to prevent the early crash or doing a little more refactoring here?

Reply:
I don't know how to test if the picker was in a working state with this fix -- the problem I was seeing was that when the app was starting in the background (for the alarm manager), nothing was working, because the plugin would crash and prevent anything else from happening. In that case, I don't actually care about the image picker.

I've no idea if it's possible for the app to then get re-used in the foreground. If so, then I agree that this wouldn't work.

Reply:
I see your point for the scenario that the app is executed in the background and then quits (and the Android APP is destroyed) it's ok to live with the plugin unregistered. So this fix at least improves some cases, we can land it now. We should probably follow-up and figure what happens if e.g the app's background code is calling startActivity and starts an activity of the same (Android) application with a FlutterView.